NEB-483: Updates to documentation on README and Javadoc (#3)
Worldpay/access-checkout-android: README.md

Add the following lines to the app level dependency list (kotlin version can be changed):

``` 

> implementation (name:'access-checkout-android-1.0.0-SNAPSHOT', ext:'aar')

Reviewer: ```suggestion
implementation (name:'access-checkout-android-1.0.0', ext:'aar')
```

### Basics

The main step required in order to use the Access Checkout SDK is the initialisation of an instance of `AccessCheckoutClient`. 
Interaction with this class will allow developers to obtain a session state from Access Worldpay services. 

This can be achieved by invoking the `init()` method of this class and by providing the required dependencies, more details provided in the following sections.

`AccessCheckoutCard` is the coordinator class between the view inputs, the (optional) validators of those inputs, and the callback of those validation results to an implementation of `CardListener`

> The SDK is fully customizable and provides default card views that implement the required interfaces out of the box. For advanced styling it is possible to override some of the default implementations or to provide fully customized implementations of the required interfaces (interfaces are described at the end of the section)

Reviewer: ```suggestion
The SDK is fully customizable and provides default card views that implement the required interfaces out of the box. For advanced styling, it is possible to override some of the default implementations or to provide fully customized implementations of the required interfaces. The interfaces are described at the end of the section.
```

cardExpiryText.cardViewListener = card
```

If taking advantage of the provided validators it is important to implement the `CardListener` interface for its callback methods.

These include all the functions invoked by the SDK at the time when the fields have finished validating on different user input events, for e.g. on each key stroke, on a field losing focus etc.

See an example of this interface being implemented in our sample application in the `MainActivity`

#### Receiving the Session State 

When the request for a session state starts, the `SessionResponseListener` is notified via the  `onRequestStarted()` callback method. 

When a result becomes available, the implementing class of `SessionResponseListener` will receive the callback holding the session state or an exception describing the error.

`onRequestFinished(sessionState: String?, error: AccessCheckoutException?)`

#### When things go wrong: `AccessCheckoutException`

If there is a problem, `SessionResponseListener` will be notified through the same `onRequestFinished(sessionState: String?, error: AccessCheckoutException?)` callback, this time with a `null` sessionState and non-null error.

> These errors can be found in the enum class `com.worldpay.access.checkout.api.AccessCheckoutException.Error`

Author: ```suggestion

The following table of errors can be found in the enum class `com.worldpay.access.checkout.api.AccessCheckoutException.Error`
```

cardExpiryText.cardViewListener = card
```

If taking advantage of the provided validators it is important to implement the `CardListener` interface for its callback methods.

These include all the functions invoked by the SDK at the time when the fields have finished validating on different user input events, for e.g. on each key stroke, on a field losing focus etc.

See an example of this interface being implemented in our sample application in the `MainActivity`

#### Receiving the Session State 

When the request for a session state starts, the `SessionResponseListener` is notified via the  `onRequestStarted()` callback method. 

When a result becomes available, the implementing class of `SessionResponseListener` will receive the callback holding the session state or an exception describing the error.

`onRequestFinished(sessionState: String?, error: AccessCheckoutException?)`

#### When things go wrong: `AccessCheckoutException`

If there is a problem, `SessionResponseListener` will be notified through the same `onRequestFinished(sessionState: String?, error: AccessCheckoutException?)` callback, this time with a `null` sessionState and non-null error.

> 

Reviewer: ```suggestion
These errors can be found in the enum class `com.worldpay.access.checkout.api.AccessCheckoutException.Error`
```

| HTTP Code | Error name | Message |
| --- | --- | --- |
| 400 | bodyIsNotJson | The body within the request is not valid json |
| 400 | bodyIsEmpty | The body within the request is empty |
| 400 | bodyDoesNotMatchSchema | The json body provided does not match the expected schema |
| 404 | resourceNotFound | Requested resource was not found |
| 404 | endpointNotFound | Requested endpoint was not found |
| 405 | methodNotAllowed | Requested method is not allowed |
| 406 | unsupportedAcceptHeader | Accept header is not supported |
| 415 | unsupportedContentType | Content-type header is not supported |

> | 500 | internalErrorOccurred |  |

Author: Lets put something here so it's not empty

> | 500 | internalErrorOccurred |  |

Reviewer: ```suggestion
| 500 | internalErrorOccurred | Internal server error |
```

> | 500 | unknownError |  |

Reviewer: ```suggestion
| 500 | unknownError | Unknown error |
```

> These errors can be found in the enum class `com.worldpay.access.checkout.api.AccessCheckoutException.Error`

Author: Move this statement above the errors table

> These errors can be found in the enum class `com.worldpay.access.checkout.api.AccessCheckoutException.Error`

Reviewer: ```suggestion

```
